src/main/scala/org/apache/spark/avro/AvroOutputWriter.scala

```diff
@@ -87,8 +85,8 @@ private[avro] class AvroOutputWriter(
         case bytes: Array[Byte] => ByteBuffer.wrap(bytes)
       }
       case ByteType | ShortType | IntegerType | LongType |
-           FloatType | DoubleType | StringType | BooleanType => identity
-      case _: DecimalType => (item: Any) => if (item == null) null else item.toString
+           FloatType | DoubleType | BooleanType => identity
+      case _: DecimalType | StringType => (item: Any) => if (item == null) null else item.toString
```

Comment:
My use case for this is serializing `Enum[_]` types.  I'm doing this as strings on the backend.  However, in my tests it was still presenting the native enum type to the native avro reader.  By moving the `StringType` out and explicitly calling `.toString` on it I'm able to get around this issue assuming the `UserDefinedType` for my enums is `StringType`.